### DISCUSSION TITLE
Pulling data from IAFD

### DISCUSSION THREAD

**Discussion by TheAnswer1313** [stashapp/stash#2642] [General]
Hello

I was wondering if there was a way to use the scrapper to just pull in all the data on a specific performer and export that to a file?

I was able to add the IAFD scrapper to stash but I can't see how actually use the scrapper.

**Comment by philpw99** [comment]
For IAFD, you do need to use the following technique with the scraper:

https://github.com/stashapp/stash/wiki/FAQ#scrape-behind-paywall

**Comment by DogmaDragon** [comment]
`IAFD.yml` requires `IAFD.py` file too. 

You also need to have Python installed on your system. `IAFD.py` also requires some dependencies:
- `py_common` folder and files inside it from https://github.com/stashapp/CommunityScrapers/tree/master/scrapers
- cloudscraper module from https://pypi.org/project/cloudscraper/ (should be installed with general Python installation)
- lxml module from https://lxml.de/installation.html#installation

You can check if you miss any modules in the Stash log after running the scraper. 

As far as using it goes, you need to provide the URL and then click the button near it to scrape the data from IAFD. 
![image](https://user-images.githubusercontent.com/103123951/176972491-d1ad8c2e-8dff-471c-b166-8846370a20a5.png)

Works for scenes, movies and performers. 

You can also search for performers by name with Scrape with... button once you click `Edit`.
![image](https://user-images.githubusercontent.com/103123951/176972826-a34c5e79-ed99-483b-9d20-079677d4f702.png)
